eng/pipelines/dotnet-monitor-release.yml

```diff
@@ -22,6 +22,144 @@ variables:
+  # - We cannot checkout the source code
+  #   - All powershell scripts were inlined in the pipeline.
+  #   - Darc installation is done via tasks instead of darc-init.
+  # - We share scripts between the validation and publish stages as variables.
+
+  # Future improvements / TODO
+  # - Move the scripts into yaml template assuming the release job allows it
+  # - Save the results of build version / release version from the validation job and use it in the release job
+  # - Validate that dotnetcli can be used for checksums
+
+- name: GetBarIdScript
+  value: |
+    $ErrorActionPreference = 'Stop'
+    Set-StrictMode -Version 2.0
+    
+    $BuildId = "$(resources.pipeline.Build.runID)"
+    $TaskVariableName = 'BarId'
+    
+    if ([String]::IsNullOrEmpty($env:System_AccessToken)) {
+        Write-Error 'System access token missing, this script needs access.'
+    }
+    
+    $tagsUri = "${env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI}${env:SYSTEM_TEAMPROJECT}/_apis/build/builds/$BuildId/tags?api-version=6.0"
+    $buildData = Invoke-RestMethod `
+        -Uri $tagsUri `
+        -Method 'GET' `
+        -Headers @{ 'accept' = 'application/json'; 'Authorization' = "Bearer ${env:System_AccessToken}" }
+    
+    Write-Verbose 'BuildData:'
+    $buildDataJson = $buildData | ConvertTo-Json
+    Write-Verbose $buildDataJson
+    
+    $barId = -1;
+    $buildData.Value | Foreach-Object {
+        if ($_.StartsWith('BAR ID - ')) {
+            if ($barId -ne -1) {
+                Write-Error 'Multiple BAR IDs found in tags.'
+            }
+            $barId = $_.SubString(9)
+        }
+    }
+    
+    if ($barId -eq -1) {
+        Write-Error 'Failed to get BAR ID from tags.'
+    }
+    
+    Write-Verbose "BAR ID: $barId"
+    
+    Write-Host "##vso[task.setvariable variable=$TaskVariableName]$barId"
+    Write-Output $barId
+  readonly: true
+- name: GetReleaseVersionScript
+  value: |
+    $ErrorActionPreference = 'Stop'
+    Set-StrictMode -Version 2.0
+    
+    $BarId = "$(BarId)"
+    $TaskVariableName = 'ReleaseVersion'
+    
+    try {
+        # Use DARC tool to get build information
+        $darcPath = "$(Agent.ToolsDirectory)/darc/darc.exe"
+        
+        if (!(Test-Path $darcPath)) {
+            Write-Error "DARC tool not found at $darcPath"
+        }
+        
+        # Get build information using DARC
+        $buildInfo = & $darcPath get-build --id $BarId --output-format json --extended --ci
+        if ($LASTEXITCODE -ne 0) {
+            Write-Error "DARC get-build command failed with exit code $LASTEXITCODE"
+        }
+        
+        $buildData = $buildInfo | ConvertFrom-Json
+        
+        [array]$matchingData = $buildData.assets | Where-Object { $_.name -match '^dotnet-monitor$' }
+        
+        if (!$matchingData -or $matchingData.Length -ne 1) {
+            Write-Error 'Unable to obtain release version'
+        }
+        
+        $version = $matchingData[0].version
```

Comment:
Assuming this is modeled off of `GetReleaseVersion`, I noticed there used to be a parameter for `$IncludeV` - do we still need this?

Reply:
Not for this context. It is being used in the nuget publish pipeline only